src/priority-queue.ts

```diff
@@ -0,0 +1,262 @@
+import {ArmorCollection} from './collection';
+import {ArmorCollectionSelector} from './selector';
+
+/**
+ * decide what to do when no comparator is passed
+ */
+
+export interface ArmorPriorityQueueOptions<T> {
+	elements?: Array<T>;
+	comparator?: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+}
+
+const defaultComparator = function (nodeOneIndex: number, nodeTwoIndex: number) {
+	if (typeof nodeOneIndex !== 'number') {
+		return false;
+	}
+	if (typeof nodeTwoIndex !== 'number') {
+		return false;
+	}
+
+	return JSON.stringify(this.elements[nodeOneIndex]) < JSON.stringify(this.elements[nodeTwoIndex]);
+};
+
+export class ArmorPriorityQueue<T> implements ArmorCollection<T> {
+	public elements: T[];
+	public comparator: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
```

Comment:
- [ ] Make it readonly

- [ ] Change from inline function type to `ArmorPriorityQueueComparator<T>`